desktop/app.js

```diff
@@ -164,20 +156,37 @@ module.exports = function main() {
     mainWindow.once('ready-to-show', mainWindow.show);
   };
 
-  const gotTheLock = app.requestSingleInstanceLock();
+  app.on('will-finish-launching', function () {
+    setTimeout(updater.ping.bind(updater), config.updater.delay);
+    app.on('open-url', function (event, url) {
+      event.preventDefault();
+      mainWindow.webContents.send('wpLogin', url);
+    });
+  });
 
-  app.on('second-instance', () => {
-    // Someone tried to run a second instance, we should focus our window.
-    if (mainWindow) {
-      if (mainWindow.isMinimized()) {
-        mainWindow.restore();
+  const gotTheLock = app.requestSingleInstanceLock();
+  if (gotTheLock) {
+    app.on('second-instance', (e, argv) => {
+      // Someone tried to run a second instance, we should focus our window.
+      // Protocol handler for win32
+      // argv: An array of the second instance’s (command line / deep linked) arguments
+      if (process.platform !== 'darwin') {
+        // Keep only command line / deep linked arguments
+        const url =
+          (argv.slice(-1) && argv.slice(-1).length && argv.slice(-1)[0]) || '';
+        mainWindow.webContents.send('wpLogin', url);
       }
-      mainWindow.focus();
-    }
-  });
 
-  if (!gotTheLock) {
-    return app.quit();
+      if (mainWindow) {
+        if (mainWindow.isMinimized()) {
+          mainWindow.restore();
+        }
+        mainWindow.focus();
+      }
+    });
+  } else {
+    app.quit();
```

Comment:
is this here because we think of this process…

 1. Click "Sign-in with WordPress.com"
 2. Close the Electron app
 3. Sign-in with the browser
 4. Browser/OS invokes new Electron app _after_ closing the window but _before_ the app process quits

?

Reply:
This doesn't have specifically to do with wpcom sign in. When we get a second instance we need to quit it so we only have one instance open. This was already in the code as:

```
if (!gotTheLock) {
    return app.quit();
```

Reply:
Looks like I was confused by the diff. What were two separate things before now are joined into one.

Would it makes sense to early-abort to keep the window management separate from the login-URL handling?

```js
const gotTheLock = app.requestSingleInstanceLock();
if (!gotTheLock) {
	return app.quit();
}

// dispatch `wpLogin` based on `argv`
if (process.platform !== 'darwin') {
	…
}
```

Reply:
Done. 

second-instance does more than "// dispatch `wpLogin` based on `argv`", it also focuses the window.